kpops/components/base_components/base_defaults_component.py

```diff
@@ -64,39 +65,35 @@ class BaseDefaultsComponent(DescConfigModel, ABC):
         description=describe_attr("enrich", __doc__),
         exclude=True,
     )
-    config: SkipJsonSchema[KpopsConfig] = Field(
-        default=...,
-        description=describe_attr("config", __doc__),
-        exclude=True,
-    )
-    handlers: SkipJsonSchema[ComponentHandlers] = Field(
-        default=...,
-        description=describe_attr("handlers", __doc__),
-        exclude=True,
-    )
+    _config: KpopsConfig = PrivateAttr()
+    _handlers: ComponentHandlers = PrivateAttr()
     validate_: SkipJsonSchema[bool] = Field(
         validation_alias=AliasChoices("validate", "validate_"),
         default=False,
         description=describe_attr("validate", __doc__),
         exclude=True,
     )
 
-    def __init__(self, **values: Any) -> None:
+    def __init__(
+        self, *, _config: KpopsConfig, _handlers: ComponentHandlers, **values: Any
+    ) -> None:
         if values.get("enrich", True):
             cls = self.__class__
-            values = cls.extend_with_defaults(**values)
-            tmp_self = cls(**values, enrich=False)
+            values = cls.extend_with_defaults(_config, **values)
+            tmp_self = cls(_config=_config, _handlers=_handlers, **values, enrich=False)
             values = tmp_self.model_dump(mode="json", by_alias=True)
-            values = cls.substitute_in_component(tmp_self.config, **values)
+            values = cls.substitute_in_component(_config, **values)
             self.__init__(
                 enrich=False,
                 validate=True,
-                config=tmp_self.config,
-                handlers=tmp_self.handlers,
+                _config=_config,
+                _handlers=_handlers,
                 **values,
             )
         else:
             super().__init__(**values)
+            self._config = _config
+            self._handlers = _handlers
```

Comment:
this feels a little hacky but seems to do the job 🫠 
(via https://github.com/microsoft/pyright/issues/7209#issuecomment-1928292955)

Reply:
nvm, does not work for Kafka Connector `_resetter` computed field, because it needs access to those attributes